### PR TITLE
Extend PoldiFitPeaks2D to include the option for Pawley type fit

### PR DESCRIPTION
**Pull request by MichaelWedel**
This fixes [#9992](http://trac.mantidproject.org/mantid/ticket/9992).

Instead of starting something new, the recently added PawleyFunction is used. The PawleyFit is slower than with individual peaks, this will be addressed in another ticket.

**Testing information**
Make sure that the (new) system test is passing, the doc tests pass. There is a new example in the doctest.

### CODE REVIEW

**Comment by MichaelWedel** [comment]
Jenkins retest this please

**Comment by MichaelWedel** [comment]
This will conflict a bit with [#475](https://github.com/mantidproject/mantid/pull/475), where I changed the error calculation. I will leave the "InProgress" label on this one until the other branch is on master to resolve the conflict.

**Comment by MichaelWedel** [comment]
All conflicts with the the changes from the other pull request have been merged and all the tests passed again.

I created a new ticket [#11461](http://trac.mantidproject.org/mantid/ticket/11461) for refactoring some parts of PoldiFitPeaks2D for the next release cycle.

File: Code/Mantid/Framework/SINQ/src/PoldiFitPeaks2D.cpp [hunk 1]
```diff
@@ -57,6 +55,29 @@ const std::string PoldiFitPeaks2D::summary() const {
...
 }
 
+/// Validate inputs for algorithm in case PawleyFit is used.
+std::map<std::string, std::string> PoldiFitPeaks2D::validateInputs() {
+  std::map<std::string, std::string> errorMap;
+
+  bool isPawleyFit = getProperty("PawleyFit");
+
+  if (isPawleyFit) {
+    Property *cellProperty = getProperty("InitialCell");
```
Comment (wdzhou) [inline Code/Mantid/Framework/SINQ/src/PoldiFitPeaks2D.cpp]: can this cause memory leak?

**Comment by MichaelWedel** [comment]
You're probably right, I think I meant to use getPointerToProperty(), which just returns the internally stored pointer. Thanks for spotting this!

**Comment by wdzhou** [comment]
Are you going to make the change in this pull request or another ticket?

**Comment by MichaelWedel** [comment]
I think it's better to fix it here, as it's a real mistake and the code freeze is coming soon. I'm pushing the update now, including the merge of master without MDEvents.

**Comment by wdzhou** [comment]
I am going to close the ticket as soon as all the checks are passed.

**Comment by MichaelWedel** [comment]
Thanks!

**Comment by wdzhou** [comment]
All tests are passed.

**Comment by wdzhou** [comment]
"This pull request is closed, but the 9992_add_poldi_pawley_function branch has unmerged commits. " Is there any commit that has not been merted?

**Comment by MichaelWedel** [comment]
Did you maybe hit the "Close" instead of the "Merge" button accidentally?

**Comment by wdzhou** [comment]
I might have made this mistake.

**Pull request merged**